### PR TITLE
fix casting

### PR DESCRIPTION
**Pull request by MixailE**

### CODE REVIEW

File: src/components/BlockchainList/PostsList.tsx [hunk 1]
```diff
@@ -1,56 +1,35 @@
...
-    >
+  if (!data) return <></>
+
+  const postsLoaded = selectedToken
+    ? data.thread.filter(
+        ({ derivativeAddress }) => derivativeAddress === PostStore.selectedToken
+      )
+    : data.thread
+
+  return postsLoaded.length > 0 ? (
```
Comment (T-Damer) [inline src/components/BlockchainList/PostsList.tsx:31]: Why can't we use `amountOfLoadedPosts` here? If those are different things, maybe we should name them differently?

File: src/components/BlockchainList/PostsList.tsx [hunk 2]
```diff
@@ -1,56 +1,35 @@
...
   const hashId = useHashParams()
-  const [, setScrolledLimit] = useState(limit)
-  const amountOfLoadedPosts = posts.length
 
-  const postsLoaded = selectedToken
-    ? posts.filter(
-        ({ derivativeAddress }) => derivativeAddress === PostStore.selectedToken
-      )
-    : posts
+  const amountOfLoadedPosts = data ? data.thread.length : 0
```
Comment (T-Damer) [inline src/components/BlockchainList/PostsList.tsx:18]: Let's put this after the `useScrollToAnchor` hook, I think hooks should go first always

File: src/components/BlockchainList/PostsList.tsx [hunk 3]
```diff
@@ -1,56 +1,35 @@
...
-        setScrolledLimit(PostStore.limit)
-      }}
-      className={scrollContainer}
-      style={{ overflow: 'hidden' }}
-      dataLength={amountOfLoadedPosts}
-      hasMore={amountOfLoadedPosts < postsAmount}
-      loader={<LoadingList text="Fetching more posts..." />}
-      endMessage={postsAmount < 3 ? <CustomizeCard /> : undefined}
-    >
+  if (!data) return <></>
```
Comment (T-Damer) [inline src/components/BlockchainList/PostsList.tsx:23]: Let's put this above `amountOfLoadedposts`, in such case you won't need to check for `data` twice

File: src/components/BlockchainList/ThreadPart.tsx [hunk 4]
```diff
@@ -115,14 +115,19 @@ export default function ({
...
         placeholder={`Reply to ${truncateMiddleIfNeeded(replyingTo, 12)}`}
       />
       {comments.map(
-        ({ timestamp, content, replier, repliedTo, replies }, index) =>
+        ({ timestamp, content, replier, repliedTo, replies, id }, index) =>
           limitThread ? (
             index < limitThread && (
               <CommentWithReplies
+                threadId={threadId}
+                id={id}
```
Comment (T-Damer) [inline src/components/BlockchainList/ThreadPart.tsx:129]: Hm, if `threadId` and `id` are the same things, why do we need both props?

Comment (MixailE) [reply src/components/BlockchainList/ThreadPart.tsx:129]: No, they are different)

File: src/hooks/useThread.ts [hunk 5]
```diff
@@ -1,19 +1,20 @@
...
-  const { threads } = useSnapshot(PostStore)
-  const thread = threads[numberId]
+export default function (threadId: number) {
+  const { threads, posts } = useSnapshot(PostStore)
+  const thread = threads[threadId]
+  const post = posts[threadId]
 
   if (!thread) {
-    fetchThread(numberId)
+    fetchThread(threadId)
```
Comment (T-Damer) [inline src/hooks/useThread.ts:10]: We should chain this logic into one `if`

**Pull request merged**